tests/Unit/Parallel/Test_AlgorithmGlobalCache.cpp

```diff
@@ -348,12 +363,130 @@ struct CheckParallelInfo {
   }
 };
 
+template <class Metavariables>
+struct CheckMemoryMonitorRelatedMethods {
+ private:
+  static inline double cache_size_ = 0.0;
+  static inline double mutable_cache_size_ = 0.0;
+
+ public:
+  using chare_type = Parallel::Algorithms::Singleton;
+  using metavariables = Metavariables;
+  using phase_dependent_action_list =
+      tmpl::list<Parallel::PhaseActions<typename Metavariables::Phase,
+                                        Metavariables::Phase::Initialization,
+                                        tmpl::list<>>>;
+  using initialization_tags = Parallel::get_initialization_tags<
+      Parallel::get_initialization_actions_list<phase_dependent_action_list>>;
+
+  static void execute_next_phase(
+      const typename Metavariables::Phase next_phase,
+      const Parallel::CProxy_GlobalCache<Metavariables>& global_cache) {
+    auto& local_cache = *Parallel::local_branch(global_cache);
+    Parallel::get_parallel_component<CheckMemoryMonitorRelatedMethods>(
+        local_cache)
+        .start_phase(next_phase);
+
+    const double time = 1.0;
+    if (next_phase == Metavariables::Phase::CallMemoryMonitorRelatedMethods) {
+      // This will broadcast to all branches of the global cache. The time is
+      // arbitrary
+      local_cache.compute_size_for_memory_monitor(time);
+
+      // This will broadcast to all branches of the mutable global cache. The
+      // time is arbitrary
+      auto mutable_global_cache_proxy =
+          local_cache.mutable_global_cache_proxy();
+      mutable_global_cache_proxy.compute_size_for_memory_monitor(global_cache,
+                                                                 time);
+
+      // Store the values for testing. The only reason this works is because we
+      // are running on 1 core, so we are always on the "local" branch of an
+      // object whether that object be a group or nodegroup. If this is run on
+      // more than one core, this will still run, but the values will be
+      // incorrect for the test
+      SPECTRE_PARALLEL_REQUIRE(Parallel::number_of_procs(local_cache) == 1);
+      cache_size_ = size_of_object_in_bytes(local_cache) / 1.0e6;
+      mutable_cache_size_ = size_of_object_in_bytes(*Parallel::local_branch(
+                                local_cache.mutable_global_cache_proxy())) /
+                            1.0e6;
+    } else if (next_phase == Metavariables::Phase::CheckMemoryMonitorFile) {
+      // This is ok because we write to disk in the previous phase so all
+      // writing is guaranted to have completed by the time we enter this phase.
```

Comment:
Sorry, since this uses a lock wouldn't it always be okay?

Reply:
I guess I'm not sure how file locks work. If the lock ensures that nothing is writing to/reading from the file, and that any writing that is occurring when the lock is called is finished first before the code continues after the lock, then I probably don't need this comment.

Reply:
Yes, that's what the lock does. Of course someone could always write without using the lock, but that's incorrect usage of the H5 infrastructure :)

Reply:
Ok I just removed the comment.